### PR TITLE
Feat/devices generate feature children

### PR DESCRIPTION
**Pull request by InteractiveTimmy**
# Pull Request

## Description

This is a rework of the [draft pull request](https://github.com/webex/webex-js-sdk/pull/1413) to rework `internal-plugin-wdm`.

This pull request contains the addition of the `FeatureModel` class and its associated tests.

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Generated associated tests for the new `FeatureModel`, and ran the plugin's tests.

**Test Configuration**:
* Node/Browser Version v8.16.0
* NPM Version v.6.4.1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/internal-plugin-devices/src/features/feature-model.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  },
+
+  /**
+   * Class object constructor. This method safely initializes the class object
+   * prior to it fully loading to allow data to be accessed and modified
+   * immediately after construction instead of initialization.
+   *
+   * @override
+   * @param {any} attrs
+   * @param {any} options
```
Comment (adamweeks) [inline packages/node_modules/@webex/internal-plugin-devices/src/features/feature-model.js]: Should this be an object? Also, can you document the default value?

File: packages/node_modules/@webex/internal-plugin-devices/src/features/feature-model.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+   * Set a property of this object to a specific value. This method utilizes
+   * code that exists within the `ampersand-state` dependency to handle
+   * scenarios in which `key = {"key": "value"}` or
+   * `key = "key", value = "value"`. Since the snippet is pulled directly from
+   * `ampersand-state`, there is no need to test both scenarios.
+   *
+   * @override
+   * @param {object | string} key - The key value, or object to be set.
+   * @param {any} value - The key value or object to set the keyed pair to.
+   * @param {any} options - The object to set the keyed pair to.
```
Comment (adamweeks) [inline packages/node_modules/@webex/internal-plugin-devices/src/features/feature-model.js]: Could this be reworded? I’m not grasping what this function does

File: packages/node_modules/@webex/internal-plugin-devices/src/features/feature-model.js (hunk 1 above)
Comment (InteractiveTimmy) [reply packages/node_modules/@webex/internal-plugin-devices/src/features/feature-model.js]: Updated the documentation.

File: packages/node_modules/@webex/internal-plugin-devices/src/features/feature-model.js (hunk 2 above)
Comment (InteractiveTimmy) [reply packages/node_modules/@webex/internal-plugin-devices/src/features/feature-model.js]: Updated the documentation.

**Pull request merged**